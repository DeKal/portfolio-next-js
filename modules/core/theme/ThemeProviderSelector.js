import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components/macro'
import {
  isChristmasDay,
  isNewYearComing,
  isTet,
  isBlogs
} from '~/modules/core/utils/helpers'
import theme from '~/modules/core/theme/styleguide/theme'
import christMasTheme from '~/modules/core/theme/styleguide/christMasTheme'
import newYearTheme from '~/modules/core/theme/styleguide/newYearTheme'
import tetTheme from '~/modules/core/theme/styleguide/tetTheme'
import blogTheme from '~/modules/core/theme/styleguide/blogTheme'
import BlogGlobalStyleCss from '~/modules/post/theme/GlobalStyleCss'
import HomeGlobalStyleCss from '~/modules/home/theme/GlobalStyleCss'
import { useRouter } from 'next/router'

const ThemeProviderSelector = ({ children }) => {
  const router = useRouter()
  const theme = useTheme(router)
  return (
    <ThemeProvider theme={theme}>
      {children}
      {selectGlobalCss(router, theme)}
    </ThemeProvider>
  )
}

ThemeProviderSelector.propTypes = {
  children: PropTypes.any
}

export default ThemeProviderSelector

const useTheme = router => {
  if (isBlogs(router.pathname)) {
    return blogTheme
  }

  if (isChristmasDay()) {
    return christMasTheme
  }

  if (isNewYearComing()) {
    return newYearTheme
  }

  if (isTet()) {
    return tetTheme
  }

  return theme
}

const selectGlobalCss = (router, theme) => {
  if (!isBlogs(router.pathname)) {
    return <HomeGlobalStyleCss theme={theme} />
  }
  return <BlogGlobalStyleCss theme={theme} />
}
