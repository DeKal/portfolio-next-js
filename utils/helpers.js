import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
