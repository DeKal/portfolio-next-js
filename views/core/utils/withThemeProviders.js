import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
