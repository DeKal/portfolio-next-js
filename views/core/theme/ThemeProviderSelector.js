import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components/macro'
import {
  isChristmasDay,
  isNewYearComing,
  isTet,
  isBlogs
} from '~/utils/helpers'
import theme from '~/theme/theme'
import christMasTheme from '~/theme/christMasTheme'
import newYearTheme from '~/theme/newYearTheme'
import tetTheme from '~/theme/tetTheme'
import blogTheme from '~/theme/blogTheme'
import GlobalStyleCss from '~/views/core/theme/GlobalStyleCss'
import { useRouter } from 'next/router'

const ThemeProviderSelector = ({ children }) => {
  const router = useRouter()
  const theme = useTheme(router)
  return (
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyleCss theme={theme} />
    </ThemeProvider>
  )
}

ThemeProviderSelector.propTypes = {
  children: PropTypes.any
}

export default ThemeProviderSelector

const useTheme = router => {
  if (isBlogs(router.pathname)) {
    return blogTheme
  }

  if (isChristmasDay()) {
    return christMasTheme
  }

  if (isNewYearComing()) {
    return newYearTheme
  }

  if (isTet()) {
    return tetTheme
  }

  return theme
}
