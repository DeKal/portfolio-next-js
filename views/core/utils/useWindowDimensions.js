import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
