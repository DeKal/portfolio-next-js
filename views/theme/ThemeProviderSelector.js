import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components/macro'
import { isChristmasDay, isNewYearComing } from '~/utils/helpers'
import theme from '~/theme/theme'
import christMasTheme from '~/theme/christMasTheme'
import newYearTheme from '~/theme/newYearTheme'

const ThemeProviderSelector = ({ children }) => {
  const theme = useTheme()
  return (
    <ThemeProvider theme={theme}>
      {children}
      <style jsx global>{`
        body:after {
          content: '';
          position: fixed;
          top: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          opacity: ${theme.images.bgPatternOpacity};
          z-index: -1;
          background: ${theme.colors.bgColor} url(${theme.images.bgPattern});
        }
      `}</style>
    </ThemeProvider>
  )
}

ThemeProviderSelector.propTypes = {
  children: PropTypes.any,
  useTheme: PropTypes.func
}

export default ThemeProviderSelector

const useTheme = () => {
  if (isChristmasDay()) {
    return christMasTheme
  }

  if (isNewYearComing()) {
    return newYearTheme
  }
  return theme
}
