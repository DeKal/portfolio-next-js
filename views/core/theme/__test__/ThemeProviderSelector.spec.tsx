import * as React from 'react'
import { shallow } from 'enzyme'
import ThemeProviderSelector from '~/views/core/theme/ThemeProviderSelector'
import * as helpers from '~/views/core/utils/helpers'
import * as Router from 'next/router'

describe('Theme Provider Selector', () => {
  describe('Selector fo normal theme', () => {
    beforeEach(() => {
      const mockUseRouter = () => ({
        pathname: '/'
      })
      // @ts-ignore 
      jest.spyOn(Router, 'useRouter').mockImplementation(mockUseRouter)
    })

    it('should render for normal day without throwing an error', function () {
      jest.spyOn(helpers, 'isChristmasDay').mockImplementation(() => false)
      jest.spyOn(helpers, 'isNewYearComing').mockImplementation(() => false)
      jest.spyOn(helpers, 'isTet').mockImplementation(() => false)
      const wrap = shallow(<ThemeProviderSelector />)
      expect(wrap).toMatchSnapshot()
    })

    it('should render for christmas day without throwing an error', function () {
      jest.spyOn(helpers, 'isChristmasDay').mockImplementation(() => true)
      jest.spyOn(helpers, 'isNewYearComing').mockImplementation(() => false)
      const wrap = shallow(<ThemeProviderSelector />)
      expect(wrap).toMatchSnapshot()
    })

    it('should render for new year day without throwing an error', function () {
      jest.spyOn(helpers, 'isChristmasDay').mockImplementation(() => false)
      jest.spyOn(helpers, 'isNewYearComing').mockImplementation(() => true)
      const wrap = shallow(<ThemeProviderSelector />)
      expect(wrap).toMatchSnapshot()
    })

    it('should render for tet without throwing an error', function () {
      jest.spyOn(helpers, 'isChristmasDay').mockImplementation(() => false)
      jest.spyOn(helpers, 'isNewYearComing').mockImplementation(() => false)
      jest.spyOn(helpers, 'isTet').mockImplementation(() => true)
      const wrap = shallow(<ThemeProviderSelector />)
      expect(wrap).toMatchSnapshot()
    })
  })

  describe('Selector fo blog theme', () => {
    beforeEach(() => {
      const mockUseRouter = () => ({
        pathname: '/blogs'
      })
      // @ts-ignore 
      jest.spyOn(Router, 'useRouter').mockImplementation(mockUseRouter)
    })
    
    it('should render for blog the,e throwing an error', function () {
      const wrap = shallow(<ThemeProviderSelector />)
      expect(wrap).toMatchSnapshot()
    })
  })

})
