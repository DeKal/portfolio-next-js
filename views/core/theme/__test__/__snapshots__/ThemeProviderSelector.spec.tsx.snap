// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Provider Selector should render for christmas day without throwing an error 1`] = `
<ThemeProvider
  theme={
    Object {
      "aboutMarginTop": "-200px",
      "colors": Object {
        "bgColor": "#fff",
        "colorAccent": "#ff8484",
        "colorDark": "#2c3e50",
        "colorDarken": "#bdc3c7",
        "colorLight": "#ecf0f1",
      },
      "defaultLineHeight": 1.57,
      "extra": Object {
        "boxShadow": "0 1px 2px -1px #2c3e50",
      },
      "font": "-apple-system, blinkmacsystemfont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
      "fontSizes": Object {
        "L": "16px",
        "M": "14px",
        "S": "12px",
        "XL": "18px",
        "XS": "10px",
        "XXL": "22px",
        "XXXL": "24px",
        "base": "14px",
      },
      "fonts": Object {
        "fontCert": "Times, serif",
        "fontCondensed": "\\"Open Sans Condensed\\", Tahoma, Verdana, Arial, sans-serif",
        "fontIp": "\\"ip\\", sans-serif",
        "fontSans": "\\"Open Sans\\", Tahoma, Verdana, Arial, sans-serif",
        "fontSerif": "Georgia, Times, \\"Times New Roman\\", serif",
        "fontTrebuchet": "Trebuchet MS, sans-serif",
      },
      "footerHeight": "50px",
      "headerPadding": "200px 10px 400px",
      "icons": Object {
        "c": "C",
        "fb": "\\\\e811",
        "goTop": "\\\\e817",
        "link": "\\\\e819",
        "linkedin": "\\\\e815",
        "location": "\\\\e806",
        "mail": "\\\\e800",
        "p": "P",
        "phone": "\\\\e81a",
        "workexp": "\\\\e804",
      },
      "images": Object {
        "background": "/images/christmas/bg-christmas.jpg",
        "bgPattern": "/images/christmas/bg-pattern-snow.png",
        "bgPatternOpacity": "0.25",
        "overlay": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAIGNIUk0AAHolAACAgwAA+f8AAIDoAABSCAABFVgAADqXAAAXb9daH5AAAAAZSURBVHjaYmCAAGM0GjfHGEMGAAAA//8DACm0ATMe8beGAAAAAElFTkSuQmCC",
        "overlayBgColor": "rgba(44, 62, 80, 0.3)",
      },
      "navContentMinHeight": "80px",
      "navHeaderHeight": "40px",
      "navHeightWhenScrollingOnMobile": "45px",
      "navPosWhenHiding": "-220px",
      "productImageRatio": "143%",
      "spaces": Object {
        "gutter": 16,
        "large": 35,
        "medium": 25,
        "small": 16,
        "xlarge": 45,
        "xsmall": 13,
        "xxsmall": 10,
        "xxxsmall": 7,
      },
    }
  }
>
  <GlobalStyleCss
    theme={
      Object {
        "aboutMarginTop": "-200px",
        "colors": Object {
          "bgColor": "#fff",
          "colorAccent": "#ff8484",
          "colorDark": "#2c3e50",
          "colorDarken": "#bdc3c7",
          "colorLight": "#ecf0f1",
        },
        "defaultLineHeight": 1.57,
        "extra": Object {
          "boxShadow": "0 1px 2px -1px #2c3e50",
        },
        "font": "-apple-system, blinkmacsystemfont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
        "fontSizes": Object {
          "L": "16px",
          "M": "14px",
          "S": "12px",
          "XL": "18px",
          "XS": "10px",
          "XXL": "22px",
          "XXXL": "24px",
          "base": "14px",
        },
        "fonts": Object {
          "fontCert": "Times, serif",
          "fontCondensed": "\\"Open Sans Condensed\\", Tahoma, Verdana, Arial, sans-serif",
          "fontIp": "\\"ip\\", sans-serif",
          "fontSans": "\\"Open Sans\\", Tahoma, Verdana, Arial, sans-serif",
          "fontSerif": "Georgia, Times, \\"Times New Roman\\", serif",
          "fontTrebuchet": "Trebuchet MS, sans-serif",
        },
        "footerHeight": "50px",
        "headerPadding": "200px 10px 400px",
        "icons": Object {
          "c": "C",
          "fb": "\\\\e811",
          "goTop": "\\\\e817",
          "link": "\\\\e819",
          "linkedin": "\\\\e815",
          "location": "\\\\e806",
          "mail": "\\\\e800",
          "p": "P",
          "phone": "\\\\e81a",
          "workexp": "\\\\e804",
        },
        "images": Object {
          "background": "/images/christmas/bg-christmas.jpg",
          "bgPattern": "/images/christmas/bg-pattern-snow.png",
          "bgPatternOpacity": "0.25",
          "overlay": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAIGNIUk0AAHolAACAgwAA+f8AAIDoAABSCAABFVgAADqXAAAXb9daH5AAAAAZSURBVHjaYmCAAGM0GjfHGEMGAAAA//8DACm0ATMe8beGAAAAAElFTkSuQmCC",
          "overlayBgColor": "rgba(44, 62, 80, 0.3)",
        },
        "navContentMinHeight": "80px",
        "navHeaderHeight": "40px",
        "navHeightWhenScrollingOnMobile": "45px",
        "navPosWhenHiding": "-220px",
        "productImageRatio": "143%",
        "spaces": Object {
          "gutter": 16,
          "large": 35,
          "medium": 25,
          "small": 16,
          "xlarge": 45,
          "xsmall": 13,
          "xxsmall": 10,
          "xxxsmall": 7,
        },
      }
    }
  />
</ThemeProvider>
`;

exports[`Theme Provider Selector should render for new year day without throwing an error 1`] = `
<ThemeProvider
  theme={
    Object {
      "aboutMarginTop": "-115px",
      "colors": Object {
        "bgColor": "#fff",
        "colorAccent": "#F0B14A",
        "colorDark": "#2c3e50",
        "colorDarken": "#bdc3c7",
        "colorLight": "#ecf0f1",
      },
      "defaultLineHeight": 1.57,
      "extra": Object {
        "boxShadow": "0 1px 2px -1px #2c3e50",
      },
      "font": "-apple-system, blinkmacsystemfont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
      "fontSizes": Object {
        "L": "16px",
        "M": "14px",
        "S": "12px",
        "XL": "18px",
        "XS": "10px",
        "XXL": "22px",
        "XXXL": "24px",
        "base": "14px",
      },
      "fonts": Object {
        "fontCert": "Times, serif",
        "fontCondensed": "\\"Open Sans Condensed\\", Tahoma, Verdana, Arial, sans-serif",
        "fontIp": "\\"ip\\", sans-serif",
        "fontSans": "\\"Open Sans\\", Tahoma, Verdana, Arial, sans-serif",
        "fontSerif": "Georgia, Times, \\"Times New Roman\\", serif",
        "fontTrebuchet": "Trebuchet MS, sans-serif",
      },
      "footerHeight": "50px",
      "headerPadding": "200px 10px 400px",
      "icons": Object {
        "c": "C",
        "fb": "\\\\e811",
        "goTop": "\\\\e817",
        "link": "\\\\e819",
        "linkedin": "\\\\e815",
        "location": "\\\\e806",
        "mail": "\\\\e800",
        "p": "P",
        "phone": "\\\\e81a",
        "workexp": "\\\\e804",
      },
      "images": Object {
        "background": "/images/newyear/bg-newyear.jpg",
        "bgPattern": "/images/newyear/bg-pattern.jpg",
        "bgPatternOpacity": "0.3",
        "overlay": "",
        "overlayBgColor": "rgba(44, 62, 80, 0.3)",
      },
      "navContentMinHeight": "80px",
      "navHeaderHeight": "40px",
      "navHeightWhenScrollingOnMobile": "45px",
      "navPosWhenHiding": "-220px",
      "productImageRatio": "143%",
      "spaces": Object {
        "gutter": 16,
        "large": 35,
        "medium": 25,
        "small": 16,
        "xlarge": 45,
        "xsmall": 13,
        "xxsmall": 10,
        "xxxsmall": 7,
      },
      "titleBgColor": "rgba(44, 25, 0, 0.6)",
    }
  }
>
  <GlobalStyleCss
    theme={
      Object {
        "aboutMarginTop": "-115px",
        "colors": Object {
          "bgColor": "#fff",
          "colorAccent": "#F0B14A",
          "colorDark": "#2c3e50",
          "colorDarken": "#bdc3c7",
          "colorLight": "#ecf0f1",
        },
        "defaultLineHeight": 1.57,
        "extra": Object {
          "boxShadow": "0 1px 2px -1px #2c3e50",
        },
        "font": "-apple-system, blinkmacsystemfont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
        "fontSizes": Object {
          "L": "16px",
          "M": "14px",
          "S": "12px",
          "XL": "18px",
          "XS": "10px",
          "XXL": "22px",
          "XXXL": "24px",
          "base": "14px",
        },
        "fonts": Object {
          "fontCert": "Times, serif",
          "fontCondensed": "\\"Open Sans Condensed\\", Tahoma, Verdana, Arial, sans-serif",
          "fontIp": "\\"ip\\", sans-serif",
          "fontSans": "\\"Open Sans\\", Tahoma, Verdana, Arial, sans-serif",
          "fontSerif": "Georgia, Times, \\"Times New Roman\\", serif",
          "fontTrebuchet": "Trebuchet MS, sans-serif",
        },
        "footerHeight": "50px",
        "headerPadding": "200px 10px 400px",
        "icons": Object {
          "c": "C",
          "fb": "\\\\e811",
          "goTop": "\\\\e817",
          "link": "\\\\e819",
          "linkedin": "\\\\e815",
          "location": "\\\\e806",
          "mail": "\\\\e800",
          "p": "P",
          "phone": "\\\\e81a",
          "workexp": "\\\\e804",
        },
        "images": Object {
          "background": "/images/newyear/bg-newyear.jpg",
          "bgPattern": "/images/newyear/bg-pattern.jpg",
          "bgPatternOpacity": "0.3",
          "overlay": "",
          "overlayBgColor": "rgba(44, 62, 80, 0.3)",
        },
        "navContentMinHeight": "80px",
        "navHeaderHeight": "40px",
        "navHeightWhenScrollingOnMobile": "45px",
        "navPosWhenHiding": "-220px",
        "productImageRatio": "143%",
        "spaces": Object {
          "gutter": 16,
          "large": 35,
          "medium": 25,
          "small": 16,
          "xlarge": 45,
          "xsmall": 13,
          "xxsmall": 10,
          "xxxsmall": 7,
        },
        "titleBgColor": "rgba(44, 25, 0, 0.6)",
      }
    }
  />
</ThemeProvider>
`;

exports[`Theme Provider Selector should render for normal day without throwing an error 1`] = `
<ThemeProvider
  theme={
    Object {
      "aboutMarginTop": "-200px",
      "colors": Object {
        "bgColor": "#fff",
        "colorAccent": "#FFA726",
        "colorDark": "#2c3e50",
        "colorDarken": "#bdc3c7",
        "colorLight": "#ecf0f1",
      },
      "defaultLineHeight": 1.57,
      "extra": Object {
        "boxShadow": "0 1px 2px -1px #2c3e50",
      },
      "font": "-apple-system, blinkmacsystemfont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
      "fontSizes": Object {
        "L": "16px",
        "M": "14px",
        "S": "12px",
        "XL": "18px",
        "XS": "10px",
        "XXL": "22px",
        "XXXL": "24px",
        "base": "14px",
      },
      "fonts": Object {
        "fontCert": "Times, serif",
        "fontCondensed": "\\"Open Sans Condensed\\", Tahoma, Verdana, Arial, sans-serif",
        "fontIp": "\\"ip\\", sans-serif",
        "fontSans": "\\"Open Sans\\", Tahoma, Verdana, Arial, sans-serif",
        "fontSerif": "Georgia, Times, \\"Times New Roman\\", serif",
        "fontTrebuchet": "Trebuchet MS, sans-serif",
      },
      "footerHeight": "50px",
      "headerPadding": "200px 10px 400px",
      "icons": Object {
        "c": "C",
        "fb": "\\\\e811",
        "goTop": "\\\\e817",
        "link": "\\\\e819",
        "linkedin": "\\\\e815",
        "location": "\\\\e806",
        "mail": "\\\\e800",
        "p": "P",
        "phone": "\\\\e81a",
        "workexp": "\\\\e804",
      },
      "images": Object {
        "background": "/images/bg.jpg",
        "bgPattern": "/images/bg-pattern.jpg",
        "bgPatternOpacity": "1",
        "overlay": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAIGNIUk0AAHolAACAgwAA+f8AAIDoAABSCAABFVgAADqXAAAXb9daH5AAAAAZSURBVHjaYmCAAGM0GjfHGEMGAAAA//8DACm0ATMe8beGAAAAAElFTkSuQmCC",
        "overlayBgColor": "rgba(44, 62, 80, 0.3)",
      },
      "navContentMinHeight": "80px",
      "navHeaderHeight": "40px",
      "navHeightWhenScrollingOnMobile": "45px",
      "navPosWhenHiding": "-220px",
      "productImageRatio": "143%",
      "spaces": Object {
        "gutter": 16,
        "large": 35,
        "medium": 25,
        "small": 16,
        "xlarge": 45,
        "xsmall": 13,
        "xxsmall": 10,
        "xxxsmall": 7,
      },
    }
  }
>
  <GlobalStyleCss
    theme={
      Object {
        "aboutMarginTop": "-200px",
        "colors": Object {
          "bgColor": "#fff",
          "colorAccent": "#FFA726",
          "colorDark": "#2c3e50",
          "colorDarken": "#bdc3c7",
          "colorLight": "#ecf0f1",
        },
        "defaultLineHeight": 1.57,
        "extra": Object {
          "boxShadow": "0 1px 2px -1px #2c3e50",
        },
        "font": "-apple-system, blinkmacsystemfont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
        "fontSizes": Object {
          "L": "16px",
          "M": "14px",
          "S": "12px",
          "XL": "18px",
          "XS": "10px",
          "XXL": "22px",
          "XXXL": "24px",
          "base": "14px",
        },
        "fonts": Object {
          "fontCert": "Times, serif",
          "fontCondensed": "\\"Open Sans Condensed\\", Tahoma, Verdana, Arial, sans-serif",
          "fontIp": "\\"ip\\", sans-serif",
          "fontSans": "\\"Open Sans\\", Tahoma, Verdana, Arial, sans-serif",
          "fontSerif": "Georgia, Times, \\"Times New Roman\\", serif",
          "fontTrebuchet": "Trebuchet MS, sans-serif",
        },
        "footerHeight": "50px",
        "headerPadding": "200px 10px 400px",
        "icons": Object {
          "c": "C",
          "fb": "\\\\e811",
          "goTop": "\\\\e817",
          "link": "\\\\e819",
          "linkedin": "\\\\e815",
          "location": "\\\\e806",
          "mail": "\\\\e800",
          "p": "P",
          "phone": "\\\\e81a",
          "workexp": "\\\\e804",
        },
        "images": Object {
          "background": "/images/bg.jpg",
          "bgPattern": "/images/bg-pattern.jpg",
          "bgPatternOpacity": "1",
          "overlay": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAIGNIUk0AAHolAACAgwAA+f8AAIDoAABSCAABFVgAADqXAAAXb9daH5AAAAAZSURBVHjaYmCAAGM0GjfHGEMGAAAA//8DACm0ATMe8beGAAAAAElFTkSuQmCC",
          "overlayBgColor": "rgba(44, 62, 80, 0.3)",
        },
        "navContentMinHeight": "80px",
        "navHeaderHeight": "40px",
        "navHeightWhenScrollingOnMobile": "45px",
        "navPosWhenHiding": "-220px",
        "productImageRatio": "143%",
        "spaces": Object {
          "gutter": 16,
          "large": 35,
          "medium": 25,
          "small": 16,
          "xlarge": 45,
          "xsmall": 13,
          "xxsmall": 10,
          "xxxsmall": 7,
        },
      }
    }
  />
</ThemeProvider>
`;

exports[`Theme Provider Selector should render for tet without throwing an error 1`] = `
<ThemeProvider
  theme={
    Object {
      "aboutMarginTop": "-111px",
      "colors": Object {
        "bgColor": "#fff",
        "colorAccent": "#FFA500",
        "colorDark": "#2c3e50",
        "colorDarken": "#bdc3c7",
        "colorLight": "#ecf0f1",
      },
      "defaultLineHeight": 1.57,
      "extra": Object {
        "boxShadow": "0 1px 2px -1px #2c3e50",
      },
      "font": "-apple-system, blinkmacsystemfont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
      "fontSizes": Object {
        "L": "16px",
        "M": "14px",
        "S": "12px",
        "XL": "18px",
        "XS": "10px",
        "XXL": "22px",
        "XXXL": "24px",
        "base": "14px",
      },
      "fonts": Object {
        "fontCert": "Times, serif",
        "fontCondensed": "\\"Open Sans Condensed\\", Tahoma, Verdana, Arial, sans-serif",
        "fontIp": "\\"ip\\", sans-serif",
        "fontSans": "\\"Open Sans\\", Tahoma, Verdana, Arial, sans-serif",
        "fontSerif": "Georgia, Times, \\"Times New Roman\\", serif",
        "fontTrebuchet": "Trebuchet MS, sans-serif",
      },
      "footerHeight": "50px",
      "headerPadding": "200px 10px 400px",
      "icons": Object {
        "c": "C",
        "fb": "\\\\e811",
        "goTop": "\\\\e817",
        "link": "\\\\e819",
        "linkedin": "\\\\e815",
        "location": "\\\\e806",
        "mail": "\\\\e800",
        "p": "P",
        "phone": "\\\\e81a",
        "workexp": "\\\\e804",
      },
      "images": Object {
        "background": "/images/tet/bg-tet.jpg",
        "bgPattern": "/images/tet/bg-pattern.jpg",
        "bgPatternOpacity": "0.1",
        "overlay": "",
        "overlayBgColor": "",
      },
      "navContentMinHeight": "80px",
      "navHeaderHeight": "40px",
      "navHeightWhenScrollingOnMobile": "45px",
      "navPosWhenHiding": "-220px",
      "productImageRatio": "143%",
      "spaces": Object {
        "gutter": 16,
        "large": 35,
        "medium": 25,
        "small": 16,
        "xlarge": 45,
        "xsmall": 13,
        "xxsmall": 10,
        "xxxsmall": 7,
      },
      "titleBgColor": "rgba(44, 25, 0, 0.2)",
    }
  }
>
  <GlobalStyleCss
    theme={
      Object {
        "aboutMarginTop": "-111px",
        "colors": Object {
          "bgColor": "#fff",
          "colorAccent": "#FFA500",
          "colorDark": "#2c3e50",
          "colorDarken": "#bdc3c7",
          "colorLight": "#ecf0f1",
        },
        "defaultLineHeight": 1.57,
        "extra": Object {
          "boxShadow": "0 1px 2px -1px #2c3e50",
        },
        "font": "-apple-system, blinkmacsystemfont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
        "fontSizes": Object {
          "L": "16px",
          "M": "14px",
          "S": "12px",
          "XL": "18px",
          "XS": "10px",
          "XXL": "22px",
          "XXXL": "24px",
          "base": "14px",
        },
        "fonts": Object {
          "fontCert": "Times, serif",
          "fontCondensed": "\\"Open Sans Condensed\\", Tahoma, Verdana, Arial, sans-serif",
          "fontIp": "\\"ip\\", sans-serif",
          "fontSans": "\\"Open Sans\\", Tahoma, Verdana, Arial, sans-serif",
          "fontSerif": "Georgia, Times, \\"Times New Roman\\", serif",
          "fontTrebuchet": "Trebuchet MS, sans-serif",
        },
        "footerHeight": "50px",
        "headerPadding": "200px 10px 400px",
        "icons": Object {
          "c": "C",
          "fb": "\\\\e811",
          "goTop": "\\\\e817",
          "link": "\\\\e819",
          "linkedin": "\\\\e815",
          "location": "\\\\e806",
          "mail": "\\\\e800",
          "p": "P",
          "phone": "\\\\e81a",
          "workexp": "\\\\e804",
        },
        "images": Object {
          "background": "/images/tet/bg-tet.jpg",
          "bgPattern": "/images/tet/bg-pattern.jpg",
          "bgPatternOpacity": "0.1",
          "overlay": "",
          "overlayBgColor": "",
        },
        "navContentMinHeight": "80px",
        "navHeaderHeight": "40px",
        "navHeightWhenScrollingOnMobile": "45px",
        "navPosWhenHiding": "-220px",
        "productImageRatio": "143%",
        "spaces": Object {
          "gutter": 16,
          "large": 35,
          "medium": 25,
          "small": 16,
          "xlarge": 45,
          "xsmall": 13,
          "xxsmall": 10,
          "xxxsmall": 7,
        },
        "titleBgColor": "rgba(44, 25, 0, 0.2)",
      }
    }
  />
</ThemeProvider>
`;
